packages/blocks/src/api/raw-handling/index.js

```diff
@@ -207,4 +208,20 @@ export default function rawHandler( { HTML = '', plainText = '', mode = 'AUTO',
 			);
 		} );
 	} ) );
+
+	// If we're allowed to return inline content and there is only one block
+	// and the original plain text content does not have any line breaks, then
+	// treat it as inline paste.
```

Comment:
This seems to be a good set of conditions to have in general, even regardless of the very specific issue this is targeting. I'm racking my brains for any unintended consequences this may have.